### PR TITLE
[BUGFIX][MER-2557] Precision not working as expected

### PR DESCRIPTION
**Pull request by simonchoxx**
[MER-2557](https://eliterate.atlassian.net/browse/MER-2557)

This PR modifies the way to check the precision of an activity input result.

To check the precision, we take the definition of `precision` given in the stack overflow link in the [ticket description](https://eliterate.atlassian.net/browse/MER-2557).

- As an example, the correct result for the following activity should be 0.36, with a precision of 2.

Captura de pantalla 2023-10-31 a la(s) 16 30 27

- Activity resolution.

https://github.com/Simon-Initiative/oli-torus/assets/16328384/5cd56647-1de4-4baa-a8fa-ad3ed14a10fe

[MER-2557]: https://eliterate.atlassian.net/browse/MER-2557?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/oli/delivery/evaluation/rule.ex [hunk 1]
```diff
@@ -142,10 +142,23 @@ defmodule Oli.Delivery.Evaluation.Rule do
...
-      |> String.split("")
-      |> Enum.filter(&is_digit?/1)
-      |> Enum.count()
+      case String.split(str, ".") do
+        ["0", decimal] ->
+          decimal
+          |> String.replace_leading("0", "")
+          |> String.split("")
+          |> Enum.filter(&is_digit?/1)
+          |> Enum.count()
```
Comment (gastonabella) [inline lib/oli/delivery/evaluation/rule.ex:151]: Nit but these functions are called in the three cases, so what do you think about moving them to a separate function? It'd result in something like:
```elixir
defp count_digits(string_number) do
  string_number
  |> String.split("")
  |> Enum.filter(&is_digit?/1)
  |> Enum.count()
end

...

case String.split(str, ".") do
  ["0", decimal] ->
    decimal
    |> String.replace_leading("0", "")
    |> count_digits()

  [_integer, _decimal] ->
    count_digits(str)

  [integer] ->
    count_digits(integer)
end
```

**Review by gastonabella — APPROVED** [review]
Looks great!

**Comment by andersweinstein** [comment]
This fix duplicates legacy "precision". While that is one approach to this issue, legacy "precision" is a very odd and unmotivated concept. What our chemistry instructors really want ultimately is correct significant figures. In discussions the OLI team decided it would be better to go straight to true significant figures on this issue, though this was not recorded on the ticket. That would break legacy parity in a couple of cases that would need to be fixed, but give chemistry and other courses the feature they really want.

The difference from legacy 'precision' arises on integers with trailing zeros: for 3400 legacy 'precision' is 4, but significant figures is 2 because trailing zeros are presumed to be there only because required as placeholders. To indicate a value of 3400 to four significant figures you must write 3400. (including a decimal point) or use scientific notation 3.400e3

**Comment by bitbldr** [comment]
Agree with @andersweinstein. We should implemented actual Significant Figures and update the instances where this might break backward compatibility with legacy, they should be few.

We should also change the label in authoring from "Precision" to "Significant Figures"

**Review by bitbldr — CHANGES_REQUESTED** [review]
See discussion from @andersweinstein around implementing sig figs

**Comment by andersweinstein** [comment]
I got into trying to code this so I've checked in my version. This still has an issue with user input of the form 3400. with decimal point, which should have 4 significant figures rather than 2. The decimal point is apparently getting dropped from the user input somewhere on the way to the precision evaluation, but have not investigated where. 

Also uncertain on desired behavior in the edge case of a string of all zero digits, e.g. 0.00. This code gives it three. It is unlikely any instructor will ever  want to test significant figures on an all zero value so it will not matter very much, but have a question out to one Chem instructor to check.

**Review by darrensiegel — APPROVED** [review]
We are landing this as-is.  Any needed tweaks for edge case handling will come in future tickets.
